test/ret/discord_client_test.exs

```diff
@@ -65,4 +67,44 @@ defmodule Ret.DiscordClientTest do
   test "moderator should be able to view restricted channel" do
     assert @moderator_user_id |> DiscordClient.has_permission?(@restricted_channel_binding, :view_channel)
   end
+
+  @tag only: true
+  test "should make a properly structured http request" do
+    parent = self()
+    mock_asserts = make_ref()
+
+    Mox.set_mox_global()
+    Mox.defmock(Ret.HttpMock, for: HTTPoison.Base)
+    Application.put_env(:ret, Ret.HttpUtils, %{:http_client => Ret.HttpMock})
+
+    Ret.HttpMock
+    |> Mox.expect(:request, 1, fn verb, url, _body, headers, _options ->
+      was_discord_api_request = url |> String.contains?(@discord_api_base)
+
+      header_keys = headers |> Enum.map(fn {k, _v} -> k |> String.downcase() end)
+      discord_request_has_ua_header = header_keys |> Enum.member?("user-agent")
+
+      send(
+        parent,
+        {mock_asserts,
+         [
+           http_verb: verb,
+           was_discord_api_request: was_discord_api_request,
+           discord_request_has_ua_header: discord_request_has_ua_header
+         ]}
+      )
+    end)
+
+    DiscordClient.has_permission?("test", %Ret.HubBinding{community_id: "test", channel_id: "test"}, :view_channel)
```

Comment:
Might not be an issue if we flush `Cachex` caches between tests, but just want to flag that the downstream request is behind a `Cachex.fetch`.